File: Arrays/Python/Non_Repeating_Element.py
class Solution:
    def firstNonRepeating(self, arr, n): 
        # Complete the function
        mp = {}
        for i in arr:
            if i in mp:
                mp[i] += 1
            else:
                mp[i] = 1
        for i in arr:
           if mp[i] == 1:
               return i
        return 0

File: Arrays/Java/Non_Repeating_Element.java
class Check{    
    public int firstNonRepeating(int arr[], int n) 
    { 
        // Complete the function
        HashMap<Integer,Integer> hm = new HashMap<>();
        for (int i=0; i<n; i++) {
            if (!hm.containsKey(arr[i])) {
                hm.put(arr[i], 1);
            }
            else {
                hm.put(arr[i], hm.get(arr[i])+1);
            }
        }
        for (int i=0; i<n; i++) {
            if (hm.get(arr[i]) == 1) {
                return arr[i];
            }
        }
        return 0;
    }  
}
